### PR TITLE
cross-compile and push images

### PR DESCRIPTION
**Pull request by deitch**
## Description

Adds the following:

* cross-compile support from `amd64` to all supported arches
* standard Makefile targets `build` and `image` which support `ARCH`
* standard Makefile target `image-all` to create all supported arches
* standard Makefile target `push` to push out a given `IMAGETAG` for an individual `ARCH`
* standard Makefile target `push-all` to push out a given `IMAGETAG` for all supported arches
* standardizes the `Dockerfile`s

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Standard support for cross-compile build, image and push for all supported arches
```

### CODE REVIEW

**Comment by deitch** [comment]
cc @fasaxc @caseydavenport @mkumatag  @djlwilder

File: Makefile [hunk 1]
```diff
@@ -59,30 +104,63 @@ build-containerized: clean vendor $(DIST)/gobgp
...
+###############################################################################
+# push images of any tag
+###############################################################################
+imagetag:
+ifndef IMAGETAG
+	$(error IMAGETAG is undefined - run using make <target> IMAGETAG=X.Y.Z)
+endif
+
+push: image imagetag
+	docker tag $(CONTAINER_NAME):latest-$(ARCH) $(CONTAINER_NAME):$(IMAGETAG)-$(ARCH)
```
Comment (fasaxc) [inline Makefile]: Still tagging in the push target here.  Please can you update this one?

**Review by fasaxc — COMMENTED** [review]
Multi-arch parts look fine to me but I don't know this build very well.

@gunjan5 or @robbrockbank any chance you could give this a once-over?

File: Makefile (hunk 1 above)
Comment (deitch) [reply Makefile]: Will do. Expect an update in ~1 hour.

Comment (deitch) [reply Makefile]: Updated

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**